code/game/objects/structures/fluff.dm

```diff
@@ -838,15 +838,21 @@ obj/structure/bars/steel
 					proceed_with_offer = TRUE
 					break
 			if(proceed_with_offer)
-				playsound(loc,'sound/items/carvty.ogg', 50, TRUE)
-				qdel(W)
+				var/list/bandits_to_benefit = list()
 				for(var/mob/player in GLOB.player_list)
-					if(player.mind)
+					if(player.mind && player.stat != DEAD) //You better be alive if you actually want your cut.
 						if(player.mind.has_antag_datum(/datum/antagonist/bandit))
-							var/datum/antagonist/bandit/bandit_players = player.mind.has_antag_datum(/datum/antagonist/bandit)
-							bandit_players.favor += donatedamnt
-							bandit_players.totaldonated += donatedamnt
-							to_chat(player, ("<font color='yellow'>[user.name] donates [donatedamnt] to the shrine! You now have [bandit_players.favor] favor.</font>"))
+							bandits_to_benefit += player.mind.has_antag_datum(/datum/antagonist/bandit)
+				if(isemptylist(bandits_to_benefit))
+					to_chat(user, span_warning("Something is wrong."))
+					return
+				playsound(loc,'sound/items/carvty.ogg', 50, TRUE)
+				qdel(W)
+				var/to_distribute = (round(donatedamnt/bandits_to_benefit.len, 0.1) + 5)	//minimum of 5 favor per item
```

Comment:
This section needs a quick adjustment. the minimum of 5 is meant to supply bandits on highpop rounds with a minimum of 5 per. However this quickly breaks down if bandits offer 1 value of something repeatedly, or if bandits are too low in pop. Please add in a check like the following.

var/to_distribute = (round(donatedamnt/bandits_to_benefit.len, 0.1) )	
if(bandits_to_benefit > 4 && donatedamnt > 20)
to_distribute = to_distribute +5